apps/emqx/src/emqx_persistent_message_ds_replayer.erl

```diff
@@ -173,13 +174,20 @@ get_last_iterator(SessionId, Stream, Ranges) ->
             Next
     end.
 
+-spec get_iterator(emqx_persistent_session_ds:id(), emqx_ds:stream()) -> emqx_ds:iterator().
 get_iterator(SessionId, Stream) ->
     Id = {SessionId, Stream},
     [#ds_iter{iter = It}] = mnesia:dirty_read(?SESSION_ITER_TAB, Id),
     It.
 
+-spec get_streams(emqx_persistent_session_ds:id()) -> [emqx_ds:stream()].
 get_streams(SessionId) ->
-    mnesia:dirty_read(?SESSION_STREAM_TAB, SessionId).
+    lists:map(
```

Comment:
Surprisingly enough, without this transformation, a weird dialyzer bug is triggered:

<details>

```
===> Error in dialyzing apps: Analysis failed with error:
{badarg,[{erl_types,t_map,
                    [[{{c,number,any,unknown},
                       mandatory,
                       {c,atom,[stream],unknown}},
                      {{c,number,any,unknown},
                       mandatory,
                       {c,number,any,unknown}},
                      {{c,number,any,unknown},
                       mandatory,
                       {c,var,'_EncapsulatedData',unknown}}],
                     none,none],
                    [{file,"erl_types.erl"},{line,1703}]},
         {erl_types,'-t_widen_to_number/1-lc$^0/1-0-',1,
                    [{file,"erl_types.erl"},{line,1607}]},
         {erl_types,'-t_widen_to_number/1-lc$^0/1-0-',1,
                    [{file,"erl_types.erl"},{line,1608}]},
         {erl_types,t_widen_to_number,1,[{file,"erl_types.erl"},{line,1608}]},
         {erl_types,'-t_widen_to_number/1-lc$^1/1-1-',1,
                    [{file,"erl_types.erl"},{line,1614}]},
         {erl_types,t_widen_to_number,1,[{file,"erl_types.erl"},{line,1615}]},
         {erl_bif_types,key_comparisons_fail,4,
                        [{file,"erl_bif_types.erl"},{line,2139}]},
         {erl_bif_types,'-type/5-fun-105-',2,
                        [{file,"erl_bif_types.erl"},{line,1246}]}]}
Last messages in the log cache:
  Reading files and computing callgraph... done in 0.29 secs
  Removing edges... done in 0.10 secs
```

</details>